### PR TITLE
Document a NewConnection footgun and the meaning of server_name

### PR DESCRIPTION
**Pull request by Ralith**
Non-obvious details identified in the course of assisting a new user.

### CODE REVIEW

**Comment by Ralith** [comment]
CI failure is unrelated:
```
error: failed to load source for a dependency on `http`

Caused by:
  Unable to update https://github.com/hyperium/http?rev=912534f1ef27d8a9050a4bd40d5ea0ee35136ea7

Caused by:
  revspec '912534f1ef27d8a9050a4bd40d5ea0ee35136ea7' not found; class=Reference (4); code=NotFound (-3)
```
@stammw any idea what's going on there?

File: quinn/src/connection.rs [hunk 1]
```diff
@@ -132,6 +132,21 @@ impl Future for ZeroRttAccepted {
 /// Components of a newly established connection
 ///
 /// Ensure `driver` runs or the connection will not work.
+///
+/// All fields of this struct, in addition to any other handles constructed later, must be dropped
+/// for a connection to be implicitly closed. If the `NewConnection` passes through a local variable
+/// in a long-running future, this might not happen when you'd expect. The easiest way to ensure
```
[the remark this answers is not in the document]
Comment (Ralith) [reply quinn/src/connection.rs]: Good point; switched to "variable".

[the remark this answers is not in the document]
Comment (djc) [reply quinn/src/connection.rs]: (Make sure you don't miss the second question, please... 👍)

[the remark this answers is not in the document]
Comment (Ralith) [reply quinn/src/connection.rs]: Added more detail there; does that clarify?

[the remark this answers is not in the document]
Comment (djc) [reply quinn/src/connection.rs]: Not really. What problem is this defending against? Have you actually tested this defense holds up in practice? In particular, I noticed that pattern destruction will also match against a reference scrutinee even if the pattern doesn't explicitly have a `&` prefix.

[the remark this answers is not in the document]
Comment (Ralith) [reply quinn/src/connection.rs]: Third attempt.

**Pull request merged**